### DISCUSSION TITLE
Problems with environment variables support 9.4

### DISCUSSION THREAD

**Discussion by CloutProject** [vercel/next.js#12815] [Help]
I recently tried to make my project work with the new environment variables support from 9.4. I removed the next.config.js file but it's not working. I have a regular .env file with variables that I want to keep private. 

I'm probably missing a concept here, but when I start my app in dev with next dev, I get the following error in my browser: "URI malformed, cannot be parsed".

**Comment by jamesmosier** [comment]
Any chance you could share some of the code? For instance where this error is being thrown from? 

Also trying logging out the environment variables being used in your code to see what is being set.

**Reply by CloutProject** [reply]
I simply cannot access any environment variable in my app. I have a .env file and I run next dev. The only way it works is when I require dotenv in a next.config.js file.

**Reply by Timer** [reply]
Our built-in environment support is disabled if you self-install `dotenv`, did you try removing it from your project all together?

**Reply by CloutProject** [reply]
Here is why! Thanks a lot for letting me know. I'm using dotenv for my migrations config file. Is there any way I can prevent it from disabling your built-in environment support?

**Reply by aorsten** [reply]
@Timer does that apply for both `dependencies` and `devDependencies`? What if I need dotenv for tests? I don't see any specific implementation of the `test` environment here: https://nextjs.org/docs/basic-features/environment-variables#exposing-environment-variables

**Reply by aorsten** [reply]
I mean, this doesn't cover the `test` environment:
```
next dev uses development
next build and next-start use production
```

I'm running tests that do not run next js, so I suppose I am wondering:
 - Can I use the .env.test environment for that (e.g. with jest) instead of dotenv?
 - or, will I need to keep using dotenv? Which brings me too to the question asked by @CloutProject

**Reply by developmentnoblenexus** [reply]
> Our built-in environment support is disabled if you self-install `dotenv`, did you try removing it from your project all together?

Works for me.
Thanks.

P.S.: It would be nice to post this on documentation section.

**Reply by vojdan** [reply]
It works for .development and .production for me, not for .test

**Comment by aorsten** [comment]
@Timer Do you have any thoughts regarding the questions raised in the discussion on the previous post?
